samcli/commands/check/bottle_necks.py

```diff
@@ -0,0 +1,72 @@
+"""
+Bottle neck questions are asked here. Data is saved in graph, but not calcualted here.
+"""
+import click
+
+from samcli.commands.check.resources.LambdaFunction import LambdaFunction
+from samcli.commands._utils.resources import AWS_LAMBDA_FUNCTION
+
+
+class BottleNecks:
+    def __init__(self, graph):
+        self._graph = graph
+        self._lambda_max_duration = 900000
+
+    def ask_entry_point_question(self):
+        entry_points = self._graph.get_entry_points()
+
+        # All entry points must be calcualted before info can be displayed
+        while entry_points:
+            entry_point_question = (
+                "We found the following resources in your application that could be the entry point for a request."
+            )
+            item_number = 1
+            for item in entry_points:
+                item_name = item.get_name()
+                entry_point_question += "\n[%i] %s" % (item_number, item_name)
+                item_number += 1
+
+            entry_point_question += "\nWhere should the simulation start?"
+            user_input = ask(entry_point_question, 1, item_number - 1)
+
+            current_entry_point = entry_points.pop(user_input - 1)
+
+            self.ask_bottle_neck_questions(current_entry_point)
+
+            self._graph.add_resource_to_analyze(current_entry_point)
+
+        click.echo("Running calculations...")
+
+    def lambda_bottle_neck_quesitons(self, lambda_function: LambdaFunction):
```

Comment:
nit: if the method is only used internally, consider prefixing it with `_` to make it a protected attribute